src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependencyNode.cs

```diff
@@ -199,7 +201,7 @@ public void AddChild(IDependencyNode childNode)
         {
             Requires.NotNull(childNode, nameof(childNode));
 
-            lock(_children)
+            lock(_childrenLock)
             {
                 _children = _children.Add(childNode);
```

Comment:
What do you mean? If there no lock here , someone else would be using old collection accessed via Children property.

---
In reply to: [99721799](https://github.com/dotnet/roslyn-project-system/pull/1469#discussion_r99721799) [](ancestors = 99721799)